src/aggregated_range_proof/dealer.rs

```diff
@@ -81,23 +106,50 @@ impl DealerAwaitingPolyCommitments {
         transcript.commit(T2.compress().as_bytes());
 
         let x = transcript.challenge_scalar();
-
-        (DealerAwaitingProofShares { n: self.n }, PolyChallenge { x })
+        let poly_challenge = PolyChallenge { x };
+
+        Ok((
+            DealerAwaitingProofShares {
+                n: self.n,
+                m: self.m,
+                value_challenge: self.value_challenge,
+                poly_challenge: poly_challenge.clone(),
+            },
+            poly_challenge,
+        ))
     }
 }
 
+#[derive(Debug)]
 pub struct DealerAwaitingProofShares {
     n: usize,
+    m: usize,
+    value_challenge: ValueChallenge,
+    poly_challenge: PolyChallenge,
 }
 
 impl DealerAwaitingProofShares {
     pub fn receive_shares(
         self,
-        transcript: &mut ProofTranscript,
         proof_shares: &Vec<ProofShare>,
         gen: &GeneratorsView,
-        y: Scalar,
-    ) -> Proof {
+        transcript: &mut ProofTranscript,
+    ) -> Result<(Proof, Vec<ProofBlame>), &'static str> {
+        if self.m != proof_shares.len() {
+            return Err("Length of proof shares doesn't match expected length m");
+        }
+
+        let mut proof_blame = Vec::new();
```

Comment:
If it's a vector, it should probably be plural, and since it's a per-share checker, should it have "share" in its name? How about `share_verifiers: Vec<ProofShareVerifier>`?

Reply:
that sounds much better! can do